src/modules/temperature.cpp

```diff
@@ -16,12 +17,42 @@ namespace modules {
   temperature_module::temperature_module(const bar_settings& bar, string name_)
       : timer_module<temperature_module>(bar, move(name_)) {
     m_zone = m_conf.get(name(), "thermal-zone", 0);
+    m_zone_type = m_conf.get(name(), "zone-type", ""s);
     m_path = m_conf.get(name(), "hwmon-path", ""s);
     m_tempbase = m_conf.get(name(), "base-temperature", 0);
     m_tempwarn = m_conf.get(name(), "warn-temperature", 80);
     set_interval(1s);
     m_units = m_conf.get(name(), "units", m_units);
 
+    if (!m_zone_type.empty()) {
+      vector<int> zone_list;
+      vector<string> thermal_list = file_util::list_files("/sys/class/thermal/");
+      for (auto &thermal: thermal_list) {
+        string zone_str = regex_replace(thermal, std::regex("thermal_zone([0-9]+)"), string("$1"), std::regex_constants::format_no_copy);
+        if (!zone_str.empty()) {
+          zone_list.push_back(stoi(zone_str));
+        }
+      }
+      sort(zone_list.begin(), zone_list.end());
+
+      bool zone_found = false;
+      string available_zones;
+      for (auto &z: zone_list) {
+        string zone_file = string_util::replace(PATH_THERMAL_ZONE_TYPE, "%zone%", to_string(z));
+        string z_zone_type = string_util::strip_trailing_newline(file_util::contents(zone_file));
+        available_zones += " " + z_zone_type;
```

Comment:
Minor: Instead of manually constructing the string, we could store the types in a vector and finally construct the string using. `string_util::join(available_zones, ", ")`

Reply:
https://github.com/polybar/polybar/pull/2752/files#diff-4ecfc5580e380c6cc659596ee5bb12d8f8df8ceea72dae9559e1494503d54691R42